pkg/storageincentives/redistributionstate_test.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package storageincentives
+
+import (
+	"context"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethersphere/bee/pkg/log"
+	erc20mock "github.com/ethersphere/bee/pkg/settlement/swap/erc20/mock"
+	"github.com/ethersphere/bee/pkg/statestore/mock"
+	transactionmock "github.com/ethersphere/bee/pkg/transaction/mock"
+	"math/big"
+	"testing"
+)
+
+func createRedistribution(t *testing.T, erc20Opts []erc20mock.Option, txOpts []transactionmock.Option) *RedistributionState {
+	t.Helper()
+	if erc20Opts == nil {
+		erc20Opts = []erc20mock.Option{
+			erc20mock.WithBalanceOfFunc(func(ctx context.Context, address common.Address) (*big.Int, error) {
+				return big.NewInt(1000), nil
+			}),
+		}
+	}
+	if txOpts == nil {
+		txOpts = []transactionmock.Option{
+			transactionmock.WithTransactionFeeFunc(func(ctx context.Context, txHash common.Hash) (*big.Int, error) {
+				return big.NewInt(1000), nil
+			}),
+		}
+	}
+	state, err := NewRedistributionState(log.Noop, common.Address{}, mock.NewStateStore(), erc20mock.New(erc20Opts...), transactionmock.New(txOpts...))
+	if err != nil {
+		t.Fatal("failed to connect")
+	}
+	return state
+}
+func TestState(t *testing.T) {
+	t.Parallel()
+	input := Status{
+		Phase:           commit,
+		IsFrozen:        false,
+		Round:           2,
+		LastWonRound:    2,
+		LastPlayedRound: 2,
+		LastFrozenRound: 2,
+		Block:           2,
+		Fees:            big.NewInt(0),
+		Reward:          big.NewInt(0),
+	}
+	want := Status{
+		Phase:           commit,
+		IsFrozen:        false,
+		Round:           2,
+		LastWonRound:    2,
+		LastPlayedRound: 2,
+		LastFrozenRound: 0,
+		Block:           2,
+		Fees:            big.NewInt(0),
+		Reward:          big.NewInt(0),
+	}
+	t.Run("all state success", func(t *testing.T) {
```

Comment:
we do not need a separate test case as you only have one input and expected output

Reply:
Right